lib/phoenix_live_view/igniter/upgrade_to_1_1.ex

```diff
@@ -39,99 +34,98 @@ defmodule Phoenix.LiveView.Igniter.UpgradeTo1_1 do
-                end)
-
-              cond do
-                index == nil ->
-                  Igniter.Code.List.prepend_to_list(zipper, :phoenix_live_view)
-
-                index == 0 ->
-                  {:ok, zipper}
+      Igniter.Project.Config.configure(
+        igniter,
+        "dev.exs",
+        app_name,
+        [endpoint_mod, :reloadable_compilers],
+        nil,
+        updater: fn zipper ->
+          if Igniter.Code.List.list?(zipper) do
+            index =
+              Igniter.Code.List.find_list_item_index(zipper, fn zipper ->
+                case Igniter.Code.Common.expand_literal(zipper) do
+                  {:ok, :phoenix_live_view} -> true
+                  _ -> false
+                end
+              end)
 
-                index > 0 ->
-                  zipper
-                  |> Igniter.Code.List.remove_index(index)
-                  |> case do
-                    {:ok, zipper} ->
-                      Igniter.Code.List.prepend_to_list(zipper, :phoenix_live_view)
+            cond do
+              index == nil ->
+                Igniter.Code.List.prepend_to_list(zipper, :phoenix_live_view)
 
-                    :error ->
-                      throw(:failed)
-                  end
-              end
-            else
-              throw(:failed)
+              index == 0 ->
+                {:ok, zipper}
+
+              index > 0 ->
+                zipper
+                |> Igniter.Code.List.remove_index(index)
+                |> case do
+                  {:ok, zipper} ->
+                    Igniter.Code.List.prepend_to_list(zipper, :phoenix_live_view)
+
+                  :error ->
+                    {:warning, warning}
+                end
             end
+          else
+            {:warning, warning}
           end
-        )
-      catch
-        :failed ->
-          Igniter.Util.Warning.warn_with_code_sample(
-            igniter,
-            """
-            You have `:reloadable_compilers` configured on your dev endpoint in config/dev.exs.
-
-            Ensure that `:phoenix_live_view` is set in there as the first entry!
-            """,
-            """
-            config :#{app_name}, #{inspect(endpoint_mod)},
-              reloadable_compilers: [:phoenix_live_view, :elixir, :app]
-            """
-          )
-      end
+        end
+      )
     else
       igniter
     end
   end
 
+  defp expected_compilers?(zipper) do
+    Igniter.Code.Function.function_call?(zipper, {Kernel, :++}) &&
+      Igniter.Code.Function.argument_equals?(zipper, 0, [:phoenix_live_view]) &&
+      Igniter.Code.Function.argument_matches_predicate?(zipper, 1, fn zipper ->
+        Igniter.Code.Function.function_call?(zipper, {Mix, :compilers})
+      end)
+  end
+
   defp maybe_update_esbuild_config(igniter) do
-    if Igniter.Util.IO.yes?(
-         "Do you want to update your esbuild configuration for colocated hooks?"
-       ) do
+    if igniter.args.options[:yes] ||
```

Comment:
I need to write this into docs or maybe even build it into the tool to enforce it, but `--yes` is important to honor to allow for scriptable installers. What this does mean though is that you might want to make this switchable w/ a CLI option